hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1305,10 +1306,11 @@ private class Emit[C](
           shapet.setup,
           datat.setup,
           rowMajort.setup)
+
         val result = Code(
           shapeAddress := shapet.value[Long],
           Code.foreach(0 until nDims) { index =>
-            shapeTuple.isMissing(index).mux[Unit](
+            shapeTuple.isFieldMissing(index).mux[Unit](
               Code._fatal[Unit](s"shape missing at index $index"),
               shapeVariables(index) := shapeTuple(index))
```

Comment:
Somehow, this line is broken. If you look a few lines up, you'll see that `shapeTuple` previously referred to a `CodePTuple` but now refers to a `PBaseStructCode`. The `PBaseStructCode` has the exact same implementation of `apply` that `CodePTuple` does, and yet when I try to compile this code I get the following stack trace:

```
E           java.lang.AssertionError: assertion failed
E           	at scala.Predef$.assert(Predef.scala:156)
E           	at is.hail.asm4s.VCode.check(Code.scala:487)
E           	at is.hail.asm4s.VCode.end(Code.scala:470)
E           	at is.hail.asm4s.Code$.apply(Code.scala:53)
E           	at is.hail.asm4s.CodeLong$.$plus$extension(Code.scala:790)
E           	at is.hail.expr.types.physical.PCanonicalBaseStruct.fieldOffset(PCanonicalBaseStruct.scala:77)
E           	at is.hail.expr.types.physical.PCanonicalBaseStruct.loadField(PCanonicalBaseStruct.scala:88)
E           	at is.hail.expr.types.physical.PBaseStructCode$$anon$2.get(PBaseStruct.scala:203)
E           	at is.hail.asm4s.package$.valueToCode(package.scala:271)
E           	at is.hail.expr.ir.Emit$$anonfun$54.apply(Emit.scala:1315)
E           	at is.hail.expr.ir.Emit$$anonfun$54.apply(Emit.scala:1312)
E           	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
E           	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
E           	at scala.collection.immutable.Range.foreach(Range.scala:160)
E           	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
E           	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
E           	at is.hail.asm4s.Code$.foreach(Code.scala:128)
E           	at is.hail.expr.ir.Emit.emit(Emit.scala:1312)
E           	at is.hail.expr.ir.Emit.emit(Emit.scala:567)
E           	at is.hail.expr.ir.Emit$$anonfun$7$$anon$2.emit(Emit.scala:358)
E           	at is.hail.expr.ir.EmitUtils$$anonfun$wrapToMethod$1.apply(Emit.scala:316)
E           	at is.hail.expr.ir.EmitUtils$$anonfun$wrapToMethod$1.apply(Emit.scala:316)
E           	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
E           	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
E           	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
E           	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
.......
```